packages/frontend/src/rtl-tests/channel.test.tsx

```diff
@@ -478,4 +488,120 @@ describe('Channel', () => {
+    const { store, runSaga } = await prepareStore(
+      {},
+      socket // Fork Nectar's sagas
+    )
+
+    const factory = await getFactory(store)
+
+    const community = await factory.create<
+    ReturnType<typeof communities.actions.addNewCommunity>['payload']
+    >('Community')
+
+    const alice = await factory.create<
+    ReturnType<typeof identity.actions.addNewIdentity>['payload']
+    >('Identity', { id: community.id, nickname: 'alice' })
+
+    window.HTMLElement.prototype.scrollTo = jest.fn()
+
+    renderComponent(
+      <>
+        <Channel />
+      </>,
+      store
+    )
+
+    await act(async () => {
+      store.dispatch(connection.actions.addInitializedCommunity(community.id))
+    })
+
+    // Log all the dispatched actions in order
+    const actions = []
+    runSaga(function* (): Generator {
+      while (true) {
+        const action = yield* take()
+        actions.push(action.type)
+      }
+    })
+
+    const messageInput = screen.getByPlaceholderText(`Message #general as @${alice.nickname}`)
+
+    // This input loses the first letter, hence the next assertion looks for a string without that.
+    userEvent.type(messageInput, 'hhello')
+
+    const isTextVisible = screen.getByText('hello')
+
+    expect(isTextVisible).toBeTruthy()
+
+    userEvent.type(messageInput, '{enter}')
+
+    // sendMessage action trigger
+    expect(actions).toMatchInlineSnapshot(`
+      Array [
+        "Messages/sendMessage",
+      ]
+    `)
+  })
+
+  it("doesn't allow to type and send message if community is not initialized", async () => {
+    const { store, runSaga } = await prepareStore(
+      {},
+      socket // Fork Nectar's sagas
+    )
+
+    const factory = await getFactory(store)
+
+    const community = await factory.create<
+    ReturnType<typeof communities.actions.addNewCommunity>['payload']
+    >('Community')
+
+    const alice = await factory.create<
+    ReturnType<typeof identity.actions.addNewIdentity>['payload']
+    >('Identity', { id: community.id, nickname: 'alice' })
+
+    window.HTMLElement.prototype.scrollTo = jest.fn()
+
+    renderComponent(
+      <>
+        <Channel />
+      </>,
+      store
+    )
+
+    // Log all the dispatched actions in order
+    const actions = []
+    runSaga(function* (): Generator {
+      while (true) {
+        const action = yield* take()
+        actions.push(action.type)
+      }
+    })
+
+    const infoMessage = screen.getByText('Initializing community. This may take a few minutes...')
+    expect(infoMessage).toBeVisible()
+
+    const messageInput = screen.getByPlaceholderText(`Message #general as @${alice.nickname}`)
+
+    // This input loses the first letter, hence the next assertion looks for a string without that.
+    userEvent.type(messageInput, 'hhello')
+
+    // We can't access value of this input by HTML, so we check if the text is visible and we expect it to throw error.
```

Comment:
I think our pattern in other cases is `expect(await screen.queryByText('hello')).toBeNull()`